src/qibolab/instruments/rfsoc/driver.py

```diff
@@ -89,6 +89,80 @@ def _try_to_execute(server_commands, host, port):
                 log.warning("Buffer full! Use shorter pulses.")
             raise e
 
+    @staticmethod
+    def convert_and_discriminate_samples(discriminated_shots, execution_parameters):
+        if execution_parameters.averaging_mode is AveragingMode.CYCLIC:
+            _, counts = np.unique(discriminated_shots, return_counts=True, axis=0)
+            freqs = counts / discriminated_shots.shape[0]
+            result = execution_parameters.results_type(freqs, discriminated_shots)
+        else:
+            result = execution_parameters.results_type(discriminated_shots)
+        return result
+
+    @staticmethod
+    def validate_input_command(
+        sequence: PulseSequence, execution_parameters: ExecutionParameters, sweep: bool
+    ):
```

Comment:
Here and above, since these methods does not make of `self` (as all `@staticmethod`s) you could even consider keeping them out of the class.

To me, the static methods are useful to bring a function together with a class, such that it is available whenever a class is available.
If, instead, you want to have a function to be used inside the class, then just define a separate function in the same module (and, if it is to be used *only* by that class, prepend an `_` to its name :P).